pkg/crawl/report.go

```diff
@@ -122,24 +226,37 @@ func (r *Reporter) Collect(ctx context.Context, path string, d fs.DirEntry, resu
 	}
 }
 
-func (r *Reporter) buildCVEMessage(versions []string) string {
-	if r.DisableCVE45105 {
-		return "CVE-2021-45046 detected"
+func (r *Reporter) matchedCVEs(versions []string) []string {
+	if len(versions) == 0 {
+		return []string{"unknown version - unknown CVE status"}
 	}
-	if cve45105VersionsOnly(versions) {
-		return "CVE-2021-45105 detected"
-	}
-	return "CVE-2021-45046 and CVE-2021-45105 detected"
-}
-
-func cve45105VersionsOnly(versions []string) bool {
-	if len(versions) == 1 && (versions[0] == "2.16.0" || versions[0] == "2.12.2") {
-		return true
+	cvesFound := make(map[string]struct{})
+	for _, version := range versions {
+		major, minor, patch, parsed := ParseLog4jVersion(version)
+		if !parsed {
+			cvesFound["unknown version - unknown CVE status"] = struct{}{}
```

Comment:
nit: Maybe `invalid version` to distinguish from the above case where versions is empty